extensions/amp-story/1.0/bookend/bookend-component.js

```diff
@@ -66,25 +60,42 @@ const textBoxComponentBuilder = new TextBoxComponent();
  */
 export let BookendComponentClass;
 
+
+/** @private @const {!Object<string, !BookendComponentClass>} */
+const builderInstances = {};
+
+
+/**
+ * @param {string} type
+ * @param {function(new:BookendComponentClass)} ctor
+ * @return {!BookendComponentClass}
+ * @restricted
+ */
+function setBuilderInstance(type, ctor) {
+  return (builderInstances[type] = builderInstances[type] || new ctor());
+}
+
+
 /**
  * Dispatches the components to their specific builder classes.
- * @param {string} componentType
+ * @param {string} type
  * @return {?BookendComponentClass}
+ * @restricted
  */
-function componentBuilderInstanceFor(componentType) {
-  switch (componentType) {
+function componentBuilderInstanceFor(type) {
+  switch (type) {
```

Comment:
Drive-by cleanup to prevent instantiating on initial story render. /cc @Enriqe

Reply:
Very cool, thanks @alanorozco !